gc/base/standard/Scavenger.cpp

```diff
@@ -1641,12 +1642,28 @@ MM_Scavenger::updateCopyScanCounts(MM_EnvironmentBase* env, uint64_t slotsScanne
 {
 	env->_scavengerStats._slotsScanned += slotsScanned;
 	env->_scavengerStats._slotsCopied += slotsCopied;
-	uint64_t updateResult = _extensions->copyScanRatio.update(env, &(env->_scavengerStats._slotsScanned), &(env->_scavengerStats._slotsCopied), _waitingCount);
+	uint64_t updateResult = _extensions->copyScanRatio.update(env, &(env->_scavengerStats._slotsScanned), &(env->_scavengerStats._slotsCopied), _waitingCount, _cachedEntryCount, _scavengeCacheScanList.getApproximateEntryCount(), &(env->_scavengerStats._copyScanUpdates));
 	if (0 != updateResult) {
 		_extensions->copyScanRatio.majorUpdate(env, updateResult, _cachedEntryCount, _scavengeCacheScanList.getApproximateEntryCount());
 	}
 }
 
+MMINLINE void
+MM_Scavenger::flushCopyScanCounts(MM_EnvironmentBase* env, bool majorFlush)
+{
+	uint64_t updateResult = 0;
+
+	if(env->_scavengerStats._slotsScanned != 0) {
+		updateResult = _extensions->copyScanRatio.update(env, &(env->_scavengerStats._slotsScanned), &(env->_scavengerStats._slotsCopied), _waitingCount, 0, 0, &(env->_scavengerStats._copyScanUpdates), true);
+	}
+
+	if(majorFlush) {
+		_extensions->copyScanRatio.flush(env);
```

Comment:
hm, shouldn't you do majorUpdate even before this flush. that would help you avoid invoking getApproximateEntryCount() on minor updates?